lesson10/js/main.js

```diff
@@ -0,0 +1,71 @@
+"use strict";
+const menuContainer = document.getElementById("js-menu_container");
+const menuLists = document.createElement("ul");
+const menuContents = [
+  {
+    to: "bookmark.html",
+    img: "images/1.png",
+    alt: "画像1",
+    text: "ブックマーク",
+  },
+  {
+    to: "message.html",
+    img: "images/2.png",
+    alt: "画像2",
+    text: "メッセージ",
+  },
+];
+
+const loading = document.getElementById("js-loading");
+
+const loadingImg = document.createElement("img");
+loadingImg.src = "images/loading-circle.gif";
+loadingImg.classList.add("el_loadingImg");
+
+function createMenuList(values) {
+  const fragment = document.createDocumentFragment();
+
+  values.forEach((value) => {
+    const listItem = document.createElement("li");
+
+    const listImg = document.createElement("img");
+    listImg.src = value.img;
+    listImg.alt = value.alt;
+    listImg.width = 40;
+    listImg.height = 40;
+
+    const listAnchor = document.createElement("a");
+    listAnchor.href = value.to;
+    listAnchor.textContent = value.text;
+
+    fragment.appendChild(listItem).appendChild(listAnchor).appendChild(listImg);
+  });
+
+  menuContainer.appendChild(menuLists).appendChild(fragment);
+}
+
+const fetchMenuContents = new Promise((resolve, reject) => {
+  loading.appendChild(loadingImg);
+
+  setTimeout(() => {
+    if (menuContents.length > 0) {
+      resolve(menuContents);
+    } else {
```

Comment:
This condition is a little strange.

Because
If there is no content, it will be an empty array, because that is normal.

Reply:
Thank you for review!

I rewrote the conditional statement. 
Changed to get an error when it is an empty array.
In this review, I was able to learn about array conditions.


[cef3ff8](https://github.com/nyattsu72/hanson-frontend-cource/pull/13/commits/cef3ff8eaeb36e2af6cefe6035e96b456dfe5d5e)

Reply:
I mean, I want  you to fix `menuContents.length > 0` to `menuContents.length >= 0`
empty array is not error.

because, Image you create services some of shop.
and then if shop goods has been sold out, API data is 

```js
{ goods: []}
```

this is not error. just sold out.
But your previous code is 

```js
 if (menuContents.length >= 0) { // fix 
      resolve([]); // this is valid
    } else {}
```
meaning to say that.

Reply:
Thank you checking the code.

I understand that 0 is not an error.
When 0 is passed, the screen becomes blank.
Added the process when 0 is passed in the try part.

Latest commit ：[ 6d57b68](https://github.com/nyattsu72/hanson-frontend-cource/pull/13/commits/6d57b683902a0016b0dacf07597ce1b54f883252)